### PR TITLE
Add test 'View raw html and text' in email themes page

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add test 'View raw html and text' in email themes page
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/08_design/03_emailThemes/03_03_viewRawHtmlAndText" URL_FO=shopUrl/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21669)

### CODE REVIEW

File: tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+  describe('View raw html and text', async () => {
+    it('should view raw html and check its URL', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'viewRawHtml', baseContext);
+
+      const newPage = await previewEmailThemesPage.viewRawHtml(page, 1);
+      await expect(newPage.url())
+        .to.contain(emailThemeName)
+        .and.to.contain('raw')
+        .and.to.contain('.html');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js:81]: It seems a little light for a test... maybe we could add more checks ? Like looking for some markup in the rendered code ?
WDYT ?

File: tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    });
+
+    it('should view raw text and check its URL', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'viewRawText', baseContext);
+
+      const newPage = await previewEmailThemesPage.viewRawText(page, 1);
+      await expect(newPage.url())
+        .to.contain(emailThemeName)
+        .and.to.contain('raw')
+        .and.to.contain('.txt');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js:108]: Same here, maybe add some checks ? Just looking at the URL and not the content seems too light for me.

File: tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js:81]: We are checking this, there is no markup.
image
I'm open for further checks but I don't know what check should be good and enough

File: tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    });
+
+    it('should get text from page and check format', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkRawTextPage', baseContext);
+
+      const pageText = await previewEmailThemesPage.getTextFromViewLayoutPage(newPage);
+      await expect(pageText)
+        .to.contain(global.FO.URL)
+        .and.to.contain('Hi');
```
Comment (SimonGrn) [inline tests/UI/campaigns/functional/BO/08_design/03_emailThemes/03_viewRawHtmlAndText.js:117]: 😆

**Review by SimonGrn — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**